Updated docs to include INFO latencystats and LATENCY HISTOGRAM (#1733)
redis/redis-doc: commands/latency-histogram.md

> The `LATENCY HISTOGRAM` command reports a cumulative distribution of latencies in the format of an histogram for each of the specified command names. 

Reviewer: ```suggestion
The `LATENCY HISTOGRAM` command reports a cumulative distribution of latencies in the format of a histogram for each of the specified command names. 
```

If no command names are specified then all commands that contain latency information will be replied.

Each reported histogram has the following fields:

* Command name.
* The total calls for that command.
* A map of time buckets:
  * Each bucket represents a latency range.
  * Each bucket covers twice the previous bucket's range.
  * Empty buckets are not printed.
  * The tracked latencies are between 1 microsecond and roughly 1 second.
  * Everything above 1 sec is considered +Inf.
  * At max there will be log2(1000000000)=30 buckets.

> This command requires the extended latency monitoring feature to be enabled (by default is enabled).

Reviewer: ```suggestion
This command requires the extended latency monitoring feature to be enabled (by default it's enabled).
```

> If you need to enable it, use 'CONFIG SET latency-tracking yes'.

Reviewer: ```suggestion
If you need to enable it, use `CONFIG SET latency-tracking yes`.
```

@examples

```
127.0.0.1:6379> LATENCY HISTOGRAM set
1# "set" => 1# "calls" => (integer) 100000
   2# "histogram_usec" => 1# (integer) 1 => (integer) 99583

>       2# (integer) 2 => (integer) 99852

Reviewer 2: Is this how redis-cli displays nested maps? (Maybe we need to do something about it in a different PR. It looks like it's misaligned in some way.)

Reviewer 3: yes. it's indeed ugly.
thing is that we don't wanna add a newline if the map contains primitive, just when it contains a compound object.
feel free to look into if if you happen to have time.

Reviewer 2: Candy: redis/redis#10170

Reviewer 3: thanks.
can you please update this PR with the improved version.
and then merge it ;)

Reviewer 2: ```suggestion
127.0.0.1:6379> LATENCY HISTOGRAM set
1# "set" =>
   1# "calls" => (integer) 100000
   2# "histogram_usec" =>
      1# (integer) 1 => (integer) 99583
      2# (integer) 2 => (integer) 99852
```

      3# (integer) 4 => (integer) 99914
      4# (integer) 8 => (integer) 99940
      5# (integer) 16 => (integer) 99968
      6# (integer) 33 => (integer) 100000
```

@return

@map-reply: specifically:

> The command returns a map where each key is a command name, and each value is a histogram object with the total calls, and time buckets inner map.

Author: @oranagra as discussed addding the extra details of the inner map and histogram object details

> @map-reply: specifically:
> 
> The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
> In RESP 2 replies, the maps are converted to arrays.

Reviewer: Until we have RESP3 docs we can't reference map replies.
```suggestion
map-reply: specifically:

The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
In RESP 2 replies, the maps are converted to [arrays](/topics/protocol#array-reply).
```

  * Empty buckets are not printed.
  * The tracked latencies are between 1 microsecond and roughly 1 second.
  * Everything above 1 sec is considered +Inf.
  * At max there will be log2(1000000000)=30 buckets.

This command requires the extended latency monitoring feature to be enabled (by default is enabled).
If you need to enable it, use 'CONFIG SET latency-tracking yes'.

@examples

RESP2 reply
```
127.0.0.1:6379> LATENCY HISTOGRAM set
1) "set"
2) 1) "calls"
   2) (integer) 100000
   3) "histogram_usec"
   4)  1) (integer) 1
       2) (integer) 99583
       3) (integer) 2
       4) (integer) 99852
       5) (integer) 4
       6) (integer) 99914
       7) (integer) 8
       8) (integer) 99940
       9) (integer) 16
      10) (integer) 99968
      11) (integer) 33
      12) (integer) 100000
```

RESP3 reply
```
127.0.0.1:6379> LATENCY HISTOGRAM set
1# "set" => 1# "calls" => (integer) 100000
   2# "histogram_usec" => 1# (integer) 1 => (integer) 99583
      2# (integer) 2 => (integer) 99852
      3# (integer) 4 => (integer) 99914
      4# (integer) 8 => (integer) 99940
      5# (integer) 16 => (integer) 99968
      6# (integer) 33 => (integer) 100000

> ```

Reviewer 3: feels a little too verbose to me.
maybe it's enough to just have one RESP version and not both.

Author: removed the RESP2 variant @oranagra given IMHO we want the most up-to-date state on the docs.

@return

@map-reply: specifically:

> The command returns an map where each key is a command name, and each value is a histogram element.

Reviewer 3: probably worth mentioning the nested map (calls and buckets here)
```suggestion
The command returns a map where each key is a command name, and each value is a histogram element.
```

@map-reply: specifically:

> The command returns a map where each key is a command name, and each value is a histogram object with the total calls, and time buckets inner map.

Reviewer 3: ```suggestion
The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
```
